cmd/kogito/deploy_data_index.go

```diff
@@ -0,0 +1,150 @@
+package main
+
+import (
+	"fmt"
+	"github.com/kiegroup/kogito-cloud-operator/pkg/apis/app/v1alpha1"
+	"github.com/kiegroup/kogito-cloud-operator/pkg/client/kubernetes"
+	resdataindex "github.com/kiegroup/kogito-cloud-operator/pkg/controller/kogitodataindex/resource"
+	"github.com/kiegroup/kogito-cloud-operator/pkg/util"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/spf13/cobra"
+)
+
+type deployDataIndexFlags struct {
+	deployCommonFlags
+	image              string
+	kafka              v1alpha1.KafkaConnectionProperties
+	infinispan         v1alpha1.InfinispanConnectionProperties
+	infinispanSasl     string
+	infinispanUser     string
+	infinispanPassword string
+}
+
+const (
+	defaultDataIndexName         = "kogito-data-index"
+	defaultInfinispanSecretName  = "kogito-data-index-infinispan-credentials"
+	defaulInfinispanUsernameKey  = "username"
+	defaultInfinispanPasswordKey = "password"
+)
+
+var (
+	deployDataIndexCmd      *cobra.Command
+	deployDataIndexCmdFlags = deployDataIndexFlags{
+		deployCommonFlags: deployCommonFlags{},
+		kafka:             v1alpha1.KafkaConnectionProperties{},
+		infinispan:        v1alpha1.InfinispanConnectionProperties{},
+	}
+)
+
+var _ = RegisterCommandVar(func() {
+	deployDataIndexCmd = &cobra.Command{
+		Use:   "deploy-data-index [flags]",
+		Short: "Deploys the Kogito Data Index Service in the given Project",
+		Long: `deploy-data-index will deploy the Data Index service to enable capturing and indexing data produced by one or more Kogito Runtime Services.
+				For more information see: https://github.com/kiegroup/kogito-runtimes/wiki/Data-Index-Service`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return deployDataIndexExec(cmd, args)
+		},
+		PreRun:  preRunF,
+		PostRun: posRunF,
+		Args: func(cmd *cobra.Command, args []string) error {
+			if len(deployDataIndexCmdFlags.infinispanUser) > 0 && len(deployDataIndexCmdFlags.infinispanPassword) == 0 {
+				return fmt.Errorf("Please set infinispan-user and infinispan-password")
```

Comment:
I think we should provide more descriptive error message here, something like "infinispan-password wasn't provided, please set both infinispan-user and infinispan-password".
Current message doesn't explicitly say what is missing.